docs/intro/overrides.rst

```diff
@@ -0,0 +1,387 @@
+    from web_poet.pages import ItemWebPage
+
+    @handle_urls("coolgadgetsite.com", overrides=GenericProductPage)
+    class CoolGadgetUsSiteProductPage(ItemWebPage):
+        def to_item(self):
+            ... # parsers here
+
+Then we could easily retrieve all Page Objects per subpackage or module like this:
+
+.. code-block:: python
+
+    from web_poet import default_registry
+
+    # We can do it per website.
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.cool_gadget_site")
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.furniture_site")
+
+    # It can also drill down to the country domains on a given site.
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.cool_gadget_site.us")
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.cool_gadget_site.fr")
+
+    # or even drill down further to the specific module.
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.cool_gadget_site.us.products")
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.cool_gadget_site.us.product_listings")
+
+    # Or simply all of Override rules ever declared.
+    rules = default_registry.get_overrides()
+
+Multiple Registry Approach
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The **package-based** approach heavily relies on how the developer organizes the
+files into intuitive hierarchies depending on the nature of the project. There
+might be cases that for some reason, a developer would want to use a **flat 
+hierarchy** like this:
+
+.. code-block::
+
+    my_page_obj_project
+    ├── __init__.py
+    ├── cool_gadget_site_us_products.py
+    ├── cool_gadget_site_us_product_listings.py
+    ├── cool_gadget_site_fr_products.py
+    ├── cool_gadget_site_fr_product_listings.py
+    ├── furniture_shop_products.py
+    └── furniture_shop_product_listings.py
+
+As such, calling ``default_registry.get_overrides_from_module()`` would not work
+on projects with a **flat hierarchy**. Thus, we can organize them using our own
+instances of the :class:`~.PageObjectRegistry` instead:
+
+.. code-block:: python
+
+    # my_page_obj_project/__init__.py
+
+    from web_poet import PageObjectRegistry
+
+    cool_gadget_registry = PageObjectRegistry()
+    cool_gadget_us_registry = PageObjectRegistry()
+    cool_gadget_fr_registry = PageObjectRegistry()
+    furniture_shop_registry = PageObjectRegistry()
+
+After declaring the :class:`~.PageObjectRegistry` instances, they can be imported
+in each of the Page Object packages like so:
+
+.. code-block:: python
+
+    # my_page_obj_project/cool_gadget_site_us_products.py
+
+    from . import cool_gadget_registry, cool_gadget_us_registry
+    from web_poet.pages import ItemWebPage
+
+    @cool_gadget_registry.handle_urls("coolgadgetsite.com", overrides=GenericProductPage)
+    @cool_gadget_us_registry.handle_urls("coolgadgetsite.com", overrides=GenericProductPage)
```

Comment:
I was having a different idea in mind when talking about multiple registries, and probably a different use case.

Let's say you have all of the following:

1. A large package, with many POs for different websites.
2. A few small packages, with POs for some specific websites.

These packages may be in separate git repos, packaged separately, etc.

There are a few use cases:

1. We want to use all of the POs defined in all these packages. This probably should be handled by web_poet.get_overrides().
2. We want to use all of the POs defined in all packages, but replace or exclude some of them (you created a better version and want to use it instead).
3. We want to use POs only from some of the packages.
4. We want to use some POs from some of the packages.

In all these cases modifying the source code of the packages may not be an option; we'd really want to select which POs to enable and which POs not to use in a flexible way.

It seems this is possible with the `get_overrides_from_module` API, but all the examples of multiple registries require making changes to the source code. I'm not sure it'd be a good practice to suggest people to create many registries and modify the source code to add more handle_urls decorators (usually with the same parameters - a lot of copy-paste).

I was thinking about a different approach for using multiple registries, something like this:

```py

registry = web_poet.PageObjectRegistry()

from some_package.pages import CoolGadgetSiteProductPage
registry.handle_urls(CoolGadgetSiteProductPage, "coolgadgetsite.com", overrides=GenericProductPage)
```

Of course, you can also have POs in your own project using the same registry, declared with a decorator.

What I don't like in the example above is that all handle_urls parameters are still duplicated

Reply:
Got it. I do see your point. Cheers for laying it out! I'll be updating this section in the docs to better showcase such scenario.

Reply:
Thanks for raising this @kmike ! I created a new dedicated section for this to explore various ways of using external packages which contain Page Objects. I can see how this would be a common requirement/usecase for the users.

> It seems this is possible with the get_overrides_from_module API, but all the examples of multiple registries require making changes to the source code. I'm not sure it'd be a good practice to suggest people to create many registries and modify the source code to add more handle_urls decorators (usually with the same parameters - a lot of copy-paste).

This section was not taking into account the use of external packages but rather exploring how to organize the Override Rules using multiple Registries in a single local project. I believe the new dedicated section named **Using Overrides from External Packages** might remove the confusion in the docs: https://github.com/scrapinghub/web-poet/pull/16/commits/bd3a88eacc0cdf69f7e94d1d428a88872a9baf50